syntax = "proto3"; // 声明protobuf的版本

package fixbug; // 声明代码所在的包（相当于c++的namespace）

option cc_generic_services = true;

message ResultCode{
    int32 errcode = 1;
    string errmsg = 2;
}

// 定义登录请求消息类型 name pwd
message LoginRequest{
    string name = 1;
    string pwd = 2;
}

// 定义登录响应消息类型
message LoginResponse{
    ResultCode result = 1;
    bool success = 2;
}


message GetFriendListsRequest{
    uint32 userid = 1;
}

message User{
    bytes name = 1;
    uint32 age = 2;
    enum Sex{
        MAN = 0;
        WOMAN = 1;
    }
    Sex sex = 3;
}

// 存储列表
message GetFriendListsResponse{
    ResultCode result = 1;
    repeated User friendsList = 2;
}

// 在protobuf中定义描述rpc方法类型-service
service UserServiceRpc{
    rpc Login(LoginRequest) returns(LoginResponse);
    rpc GetFriendLists(GetFriendListsRequest) returns(GetFriendListsResponse);
}